In [1]:
import pandas as pd

dataset_path = '../data/raw/housing-price-train.csv'
df_raw = pd.read_csv(dataset_path)
df_eda = df_raw.copy()

In [2]:
print(df_eda.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

In [3]:
# Summary of null values in each column
null_summary = (
    df_eda.isna() # Check for nulls
    .mean() # Calculate mean (proportion of nulls)
    .mul(100) # Convert to percentage
    .round(2) # Round to 2 decimal places
    .sort_values(ascending=False) # Sort descending
)

# Display only columns containing null values
null_columns_percentages = null_summary[null_summary > 0]

print(null_columns_percentages)

PoolQC          99.52
MiscFeature     96.30
Alley           93.77
Fence           80.75
MasVnrType      59.73
FireplaceQu     47.26
LotFrontage     17.74
GarageQual       5.55
GarageFinish     5.55
GarageType       5.55
GarageYrBlt      5.55
GarageCond       5.55
BsmtFinType2     2.60
BsmtExposure     2.60
BsmtCond         2.53
BsmtQual         2.53
BsmtFinType1     2.53
MasVnrArea       0.55
Electrical       0.07
dtype: float64


In [ ]:
# Select only the specified columns for prediction
df_selected = df_eda[['MSZoning', 'YrSold', 'GarageType', 'SalePrice']].copy()

# Optional: Display the new DataFrame info
print(df_selected.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   MSZoning    1460 non-null   object
 1   YrSold      1460 non-null   int64 
 2   GarageType  1379 non-null   object
 3   SalePrice   1460 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 45.8+ KB
None


In [5]:
# Check unique values and counts in 'payment_method' column
print(df_selected['GarageType'].value_counts())

# Alternatively, just list unique values
print(df_selected['GarageType'].unique())

GarageType
Attchd     870
Detchd     387
BuiltIn     88
Basment     19
CarPort      9
2Types       6
Name: count, dtype: int64
['Attchd' 'Detchd' 'BuiltIn' 'CarPort' nan 'Basment' '2Types']


In [6]:
# Impute missing values in 'Payment Method' with the mode
mode_value = df_selected['GarageType'].mode()[0]
df_selected['GarageType'].fillna(mode_value, inplace=True)

# Verify imputation by checking nulls again
print(df_selected['GarageType'].isna().sum())

0


/tmp/ipykernel_43731/2469393419.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_selected['GarageType'].fillna(mode_value, inplace=True)


In [7]:
# Check for duplicate rows
duplicate_count = df_selected.duplicated().sum()
print(f"Number of duplicate rows: {duplicate_count}")

# To view the duplicate rows (optional)
if duplicate_count > 0:
    print(df_selected[df_selected.duplicated()])

Number of duplicate rows: 178
     MSZoning  YrSold GarageType  SalePrice
116        RL    2009     Attchd     139000
121        RM    2007     Detchd     100000
134        RL    2006     Attchd     180000
138        RL    2008     Attchd     230000
141        RL    2006     Attchd     260000
...       ...     ...        ...        ...
1434       RL    2006     Attchd     160000
1435       RL    2008     Attchd     174000
1445       RL    2007     Detchd     129000
1447       RL    2007     Attchd     240000
1455       RL    2007     Attchd     175000

[178 rows x 4 columns]


## Observations from Data Preparation

- **Null Values**: Columns with missing data include 'GarageType' (5.5% nulls). Imputation was applied using the mode (most frequent value: 'Attchd').
- **Duplicates**: No abnormal duplicates were found.